libs/components/tabs/src/lib/modules/tabs/tabset-nav-button.component.ts

```diff
@@ -23,126 +20,174 @@ const buttonTypePrevious = 'previous';
   selector: 'sky-tabset-nav-button',
   templateUrl: './tabset-nav-button.component.html',
 })
-export class SkyTabsetNavButtonComponent implements AfterViewInit {
+export class SkyTabsetNavButtonComponent implements OnDestroy {
+  /**
+   * Specifies the tabset wizard component to associate with the nav button.
+   * @required
+   */
   @Input()
-  public tabset: SkyTabsetComponent;
+  public set tabset(value: SkyTabsetComponent) {
+    this.#_tabset = value;
 
-  @Input()
-  public buttonType: string;
+    if (this.#currentTabsetSub) {
+      this.#currentTabsetSub.unsubscribe();
+    }
+
+    if (value) {
+      if (this.#_buttonType === buttonTypeFinish) {
+        this.#_tabset.hasFinishButton = true;
+      }
+
+      this.#currentTabsetSub = this.#_tabset.activeChange
+        .pipe(distinctUntilChanged(), takeUntil(this.#ngUnsubscribe))
+        .subscribe((index: SkyTabIndex) => {
+          this.#activeSkyTabIndex = index;
+          this.#tabCount = this.#_tabset.tabs.length;
+          this.#updateTabToSelect();
+          this.#updateButtonVisibility();
+          this.#updateButtonProperties();
+        });
+    } else {
+      this.#logger.error(
+        'The SkyTabsetNavButtonComponent requires a reference to the SkyTabsetComponent it controls.'
+      );
+    }
+  }
 
+  /**
+   * Specifies the type of nav button to include.
+   * @required
+   */
   @Input()
-  public set buttonText(value: string) {
-    this._buttonText = value;
+  public get buttonType(): SkyTabsetNavButtonType {
+    return this.#_buttonType;
   }
 
-  public get buttonText(): string {
-    if (this._buttonText) {
-      return this._buttonText;
-    }
+  public set buttonType(value: SkyTabsetNavButtonType) {
+    this.#_buttonType = value;
+    this.#updateTabToSelect();
+    this.#updateButtonProperties();
+    this.#updateButtonVisibility();
 
-    switch (this.buttonType) {
-      case buttonTypePrevious:
-        return this.previousButtonText;
-      case buttonTypeNext:
-        return this.nextButtonText;
-      /* istanbul ignore next */
-      default:
-        return '';
+    if (value === buttonTypeFinish) {
+      this.type = 'submit';
+    } else {
+      this.type = 'button';
     }
   }
 
+  /**
+   * Specifies the label to display on the nav button. The following are the defaults for each `buttonType`.
+   * `next` = "Next", `previous` = "Previous", `finish` = "Finish"
+   */
+  @Input()
+  public buttonText?: string | undefined;
+
+  /**
+   * Indicates whether to disable the nav button.
+   * Defaults to the disabled state of the next tab for `next`, the existence of a previous tab for `previous`,
+   * and false for `finish`.
+   */
+  @Input()
+  public set disabled(value: boolean) {
+    this.#_disabled = value;
```

Comment:
We should probably set a separate public property in the setter here and update it when `buttonType` updates or when `disabled` updates and then use that separate property in the template. That way we don't have logic in the getter but the getter also doesn't return something different than what was set. We do similar in other places that have defaults such as `ariaRoleOrDefault` in some places.

Reply:
unfortunately the nav buttons rely on it being a getter. Every time something changes in the wizard form (i.e. user input), we do need to re-check if its disabled since it relies on the tab's disabled property and we don't emit any event when that changes